website/docs/configuration/expressions.html.md

```diff
@@ -16,195 +16,204 @@ references to data exported by resources, arithmetic, conditional evaluation,
 and a number of built-in functions.
 
 Expressions can be used in a number of places in the Terraform language,
-but some contexts place restrictions on which expression constructs are allowed,
-such as requiring a literal value of a particular type, or forbidding
-references to resource attributes. The other pages in this section describe
-the contexts where expressions may be used and which expression features
-are allowed in each case.
+but some contexts limit which expression constructs are allowed,
+such as requiring a literal value of a particular type or forbidding
+references to resource attributes. Each language feature's documentation
+describes any restrictions it places on expressions.
 
-The following sections describe all of the features of the configuration
-syntax.
+You can experiment with the behavior of Terraform's expressions from
+the Terraform expression console, by running
+[the `terraform console` command](/docs/commands/console.html).
+
+The rest of this page describes all of the features of Terraform's
+expression syntax.
 
 ## Types and Values
 
 The result of an expression is a _value_. All values have a _type_, which
 dictates where that value can be used and what transformations can be
 applied to it.
 
-A _literal expression_ is an expression that directly represents a particular
-constant value.
-
-Expressions are most commonly used to set the values of arguments to resources
-and to child modules. In these cases, the argument itself has an expected
-type and so the given expression must produce a value of that type. Where
-possible, Terraform will automatically convert values from one type to another
-in order to produce the expected type. If this isn't possible, Terraform will
-produce a type mismatch error and you must update the configuration with
-a more suitable expression.
-
-This section describes all of the value types in the Terraform language, and
-the literal expression syntax that can be used to create values of each
-type.
-
-### Primitive Types
-
-A _primitive_ type is a simple type that isn't made from any other types.
-The available primitive types in the Terraform language are:
-
-* `string`: a sequence of Unicode characters representing some text, such
-  as `"hello"`.
+The Terraform language uses the following types for its values:
 
+* `string`: a sequence of Unicode characters representing some text, like
+  `"hello"`.
 * `number`: a numeric value. The `number` type can represent both whole
-  numbers like `15` and fractional values such as `6.283185`.
-
+  numbers like `15` and fractional values like `6.283185`.
 * `bool`: either `true` or `false`. `bool` values can be used in conditional
   logic.
+* `list` (or `tuple`): a sequence of values, like
+  `["us-west-1a", "us-west-1c"]`. Elements in a list or tuple are identified by
+  consecutive whole numbers, starting with zero.
+* `map` (or `object`): a group of values identified by named labels, like
+  `{name = "Mabel", age = 52}`.
```

Comment:
This is the spot where I started conflating the similar complex types more aggressively, pushing the full details out into the type constraints page.